packages/features/ee/workflows/components/WorkflowDetailsPage.tsx

```diff
@@ -36,19 +38,40 @@ export default function WorkflowDetailsPage(props: Props) {
 
   const eventTypeOptions = useMemo(
     () =>
-      data?.eventTypeGroups.reduce(
-        (options, group) => [
-          ...options,
-          ...group.eventTypes.map((eventType) => ({
-            value: String(eventType.id),
-            label: eventType.title,
-          })),
-        ],
-        [] as Option[]
-      ) || [],
+      data?.eventTypeGroups
+        .filter((eventType) => {
+          if (!teamId && !eventType.teamId) {
+            return eventType;
+          }
+          if (teamId === eventType.teamId) {
+            return eventType;
+          }
+        })
+        .reduce(
+          (options, group) => [
+            ...options,
+            ...group.eventTypes.map((eventType) => ({
+              value: String(eventType.id),
+              label: eventType.title,
+            })),
+          ],
+          [] as Option[]
+        ) || [],
```

Comment:
We can transform and filter in a single iteration as well:

```suggestion
      data?.eventTypeGroups
        .reduce(
          (options, group) => {
            /** only show event types that belong to team or user */
            if (!(!teamId && !group.teamId)) return options;
            if (teamId !== group.teamId) {
              return options;
            }
            return [
            ...options,
            ...group.eventTypes.map((eventType) => ({
              value: String(eventType.id),
              label: eventType.title,
            })),
          ]},
          [] as Option[]
        ) || [],
```